pkg/cloud/azure/services/compute/virtualmachines.go

```diff
@@ -78,31 +78,91 @@ func (s *Service) createVM(machine *actuators.MachineScope, bootstrapToken, kube
 
 	input := &v1alpha1.VM{}
 
+	startupScript, err := s.getVMStartupScript(machine, bootstrapToken)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error while trying to get VM startupscript for machine name %s in cluster %s", machine.Name(), s.scope.Name())
+	}
+	input.StartupScript = startupScript
+	// TODO: Set ssh key
+
+	resourcesSvc := resources.NewService(s.scope)
+	err = resourcesSvc.ValidateDeployment(machine.Machine, s.scope.ClusterConfig, machine.MachineConfig, input.StartupScript)
+	if err != nil {
+		return nil, fmt.Errorf("error validating deployment: %v", err)
+	}
+
+	deploymentsFuture, err := resourcesSvc.CreateOrUpdateDeployment(machine.Machine, s.scope.ClusterConfig, machine.MachineConfig, input.StartupScript)
+	if err != nil {
+		return nil, fmt.Errorf("error creating or updating deployment: %v", err)
+	}
+
+	err = resourcesSvc.WaitForDeploymentsCreateOrUpdateFuture(*deploymentsFuture)
+	if err != nil {
+		return nil, fmt.Errorf("error waiting for deployment creation or update: %v", err)
+	}
+
+	deployment, err := resourcesSvc.GetDeploymentResult(*deploymentsFuture)
+	// Work around possible bugs or late-stage failures
+	if deployment.Name == nil || err != nil {
+		return nil, fmt.Errorf("error getting deployment result: %v", err)
+	}
+
+	vm, err := s.scope.VM.Get(s.scope.Context, machine.ClusterConfig.ResourceGroup, machine.Name(), "")
+	if err != nil {
+		return nil, err
+	}
+
+	record.Eventf(machine.Machine, "CreatedVM", "Created new %s vm with id %q", machine.Role(), vm.ID)
+	return converters.SDKToVM(vm), nil
+}
+
+// DeleteVM deletes the virtual machine.
+func (s *Service) DeleteVM(resourceGroup string, name string) (compute.VirtualMachinesDeleteFuture, error) {
+	return s.scope.VM.Delete(s.scope.Context, resourceGroup, name)
+}
+
+// MachineExists will return whether or not a machine exists.
+func (s *Service) MachineExists(machine *actuators.MachineScope) (bool, error) {
+	var err error
+	var instance *v1alpha1.VM
+
+	if machine.MachineStatus.VMID != nil {
+		instance, err = s.VMIfExists(machine.Name())
+	}
+
+	if err != nil {
+		return false, errors.Wrapf(err, "failed to lookup machine %q", machine.Name())
+	}
+	return instance != nil, nil
+}
+
+func (s *Service) getVMStartupScript(machine *actuators.MachineScope, bootstrapToken string) (string, error) {
+	var startupScript string
+
```

Comment:
This looks like it belongs in a different PR than one for updating gloangci-lint